docs/typing-indicators.md

```diff
@@ -0,0 +1,146 @@
+# Typing indicators
+
+Zulip supports a feature called "typing indicators."
+
+Typing indicators are are status messages that tell you when
+another user is composing a message to you.  Zulip' typing UI
+is similar to what you see in other chat/text systems.
+
+This document describes how we implement the feature in Zulip,
+and our main audience is developers who want to understand the
+system and possibly improve it.  This document assumes that the
+client is our web app, but any client can play along with this
+protocol.
+
+Right now typing indicators are only used in "Private Messages"
+views.  We implemented them for streams in a very early version of
+the feature, but users found them distracting.  We may re-visit
+this decision, particularly for private streams or streams with
+fewer invitees.
```

Comment:
We actually never implemented them for streams; I just cut streams from the initial implementation for simplicity.